core/genesis.go

```diff
@@ -485,6 +485,9 @@ func (g *Genesis) ToBlock() *types.Block {
 		if g.Difficulty == nil && g.Mixhash == (common.Hash{}) {
 			head.Difficulty = params.GenesisDifficulty
 		}
+	} else if g.Difficulty == nil {
+		// In the case of migrated chains we ensure a zero rather than nil difficulty.
+		head.Difficulty = new(big.Int)
```

Comment:
Mh, after the issue Karl found I wonder if this will bring trouble in the future: https://github.com/celo-org/op-geth/pull/153

With this change, `isMerge` will be true for all blocks. 🤔

Reply:
Could you elaborate? The linked issue seems to be about setting the Random field in the block context for vm execution. I don't see how that connects with setting difficulty here?

Reply:
Ok got it, in `NewEvmBlockContext` `Random` is set to the header's mix hash if difficulty is equal to zero. It seems like this shouldn't affect us since we are not supporting execution for historical blocks, so we shouldn't end up in this code path. If we didn't set this to zero then we'd run into a nil pointer exception.